### DISCUSSION TITLE
Why there is no `__init__` in the paginator classes??

### DISCUSSION THREAD

**Discussion by ZeyadMoustafaKamal** [encode/django-rest-framework#9424] [Ideas & Suggestions]
Well, this is weird to me as I have no idea why this hasn't already been implemented. What I mean is that it's normal to think about. Anyway let's say I want to add something like the `page_size` in the `PageNumberPagination` what will I have to do ?? I will need to subclass the `PageNumberPagination` or add it in the defaults in the `settings.py` but here is the problem. what if I had different pagination parameters depending on something? like the notifications, the comments, the posts ...etc. All of them will be different; this way, I will need to subclass the paginator class for every use case. 

So I want to do one of the two things. maybe provide a way to add some class variables in the views or viewsets that control these things or allow to pass it to the '__init__' (or both) also I am ready to implement any of those and open a PR. I want your suggestions for this. what do you think??

**Comment by cclauss** [comment]


**Reply by ZeyadMoustafaKamal** [reply]
No. I am not talking about a package. I am talking about `__init__` the method. like this

```Python
class Foo:
    def __init__(self, bar):
        self.bar = bar
my_foo = Foo("Something")

```

I think that you will know what I am talking about if you just read it.

**Reply by cclauss** [reply]
Sorry.  I misread.

**Reply by ZeyadMoustafaKamal** [reply]
@cclauss not a problem. but what do you think?? should I implement it or go back?? I just need any suggestion
